evadb/optimizer/rules/rules.py

```diff
@@ -561,26 +571,34 @@ def _exists_predicate(opr):
 
         # Get column catalog entry and function_signature.
         column_catalog_entry = tv_expr.col_object
-        function_signature = (
-            None
-            if isinstance(base_func_expr, TupleValueExpression)
-            else base_func_expr.signature()
-        )
 
-        # Get index catalog. Check if an index exists for matching
-        # function signature and table columns.
-        index_catalog_entry = (
-            catalog_manager().get_index_catalog_entry_by_column_and_function_signature(
+        # Only check the index existence when building on EvaDB data.
+        if not is_postgres_data_source:
+            # Get function_signature.
+            function_signature = (
+                None
+                if isinstance(base_func_expr, TupleValueExpression)
+                else base_func_expr.signature()
+            )
+
```

Comment:
Does this mean that for pgvector, `CREATE INDEX test_index ON postgres.mytable (FUNC(data)) USING PGVECTOR` will not work?

Reply:
The plan for this PR is to add a feature that index can be directly created on the raw table. I think the function expression use case is important, but doing it together makes this PR too complicated so will file a separate one later.